packages/legend-art/src/components/panel/PanelHeader.tsx

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { capitalize } from '@finos/legend-shared';
+import { clsx } from 'clsx';
+import { observer } from 'mobx-react-lite';
+
+export const PanelHeaderActionItem: React.FC<{
+  tip: string;
+  className?: string;
+  disabled?: boolean;
+  children: React.ReactNode;
+  onClick?: () => void;
+}> = (props) => {
+  const { className, onClick, children, disabled, tip } = props;
+  return (
+    <button
+      className={clsx('panel__header__action', className)}
+      disabled={Boolean(disabled)}
+      onClick={onClick}
+      title={tip}
+    >
+      {children}
+    </button>
+  );
+};
+
+export const PanelHeader: React.FC<{
+  title: string;
+  children?: React.ReactNode;
+  className?: string;
+}> = (props) => {
+  const { title, children } = props;
+  return (
+    <div className="panel__header">
+      <div className="panel__header__title">
+        <div className="panel__header__title__label">{title.toLowerCase()}</div>
+      </div>
+      {children && <div className="panel__header__actions">{children}</div>}
+    </div>
+  );
+};
+
+export const PanelForm: React.FC<{
+  children: React.ReactNode;
+  className?: string;
+}> = (props) => {
+  const { children } = props;
+  return <div className="panel__content__form">{children}</div>;
+};
+
+export const PanelFormRow: React.FC<{
+  children: React.ReactNode;
+  className?: string;
+}> = (props) => {
+  const { children } = props;
+  return <div className="panel__content__form__row">{children}</div>;
+};
+
+export const PanelTextEditor = observer(
+  (props: {
+    propertyName: string;
+    description?: string;
+    padding?: boolean;
```

Comment:
I'm iffy about `padding`, when do you need it?

Reply:
With the padding there is another editor that has text with 0 margins on left and right so it looks more aligned with padding margins being 0 0.5? But it is not needed for mapper schemas and tables since they are the only things here, so removed in mapper editor 👍